node.go

```diff
@@ -453,8 +441,8 @@ func (n *Node) handleStateMessage(msg messages.State) (newMessage bool) {
 	curr, exist := n.peerStates[msg.NodeName]
 	if exist && msg.Time <= curr.Time {
 		// Ignore a state message if we have the same or a newer one.
-		return false
-	} else if exist && msg.NodeName == n.cfg.Name {
```

Comment:
There was two problems here:

1. `n.peerStates` will be empty when we first start up, so the exist check will fail. The condition in nodeDelegate.MergeRemoteState doesn't have the `exist` check, so I'm not sure what I was thinking here (it looks like it was just a bug)

2. Returning `true` caused the original msg to be rebroadcasted, which is wrong! We want to also return the modified message so we broadcast that instead.